src/nupic/research/temporal_memory.py

```diff
@@ -362,35 +261,37 @@ def activatePredictedColumn(connections, excitedColumn, learn,
 
     cellsToAdd = []
     cell = None
-    for active in excitedColumn["activeSegments"]:
-      newCell = not cell == connections.cellForSegment(active)
+    for active in activeSegments:
+      newCell = cell != active.segment.cell
       if newCell:
-        cell = connections.cellForSegment(active)
+        cell = active.segment.cell
         cellsToAdd.append(cell)
 
       if learn:
         TemporalMemory.adaptSegment(connections, prevActiveCells,
                                     permanenceIncrement, permanenceDecrement,
-                                    active)
+                                    active.segment)
 
     return cellsToAdd
 
 
   @staticmethod
-  def burstColumn(cellsPerColumn, connections, excitedColumn,
-                  learn, initialPermanence, maxNewSynapseCount,
-                  permanenceDecrement, permanenceIncrement,
+  def burstColumn(cellsPerColumn, column, connections,
+                  initialPermanence, learn, matchingSegments,
+                  maxNewSynapseCount, permanenceDecrement, permanenceIncrement,
                   prevActiveCells, prevWinnerCells, random):
     """ Activates all of the cells in an unpredicted active column,
     chooses a winner cell, and, if learning is turned on, either adapts or
     creates a segment. growSynapses is invoked on this segment.
 
     @param cellsPerColumn      (int)    Number of cells per column
+    @param column              (int)    Index of bursting column
     @param connections         (Object) Connections instance for the tm
-    @param excitedColumn       (dict)   Excited Column instance from
-                                        excitedColumnsGenerator
-    @param learn               (bool)   Whether or not learning is enabled
     @param initialPermanence   (float)  Initial permanence of a new synapse.
+    @param learn               (bool)   Whether or not learning is enabled
+    @param matchingSegments    (list)   A list of SegmentOverlap objects for the
+                                        column compute is operating on that are
+                                        matching
```

Comment:
This is now an iterable, and the value is None sometimes. It's worth documenting that. Same with the other method docstrings.